packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -245,13 +253,49 @@ export class FungibleTokenTransferController implements ReactiveController {
             !this.supportedResources.includes(route.resource))
       )
       .map((route) => route.resource);
-    //unselect destination if equal to source network or isn't in list of available destination networks
-    if (
-      this.destinationNetwork?.id === sourceNetwork.id ||
-      !this.supportedDestinationNetworks.includes(this.destinationNetwork!)
-    ) {
+
+    // unselect destination if equal to source network or isn't in list of available destination networks
+    if (this.destinationNetwork?.id === sourceNetwork.id) {
       this.destinationNetwork = undefined;
+      this.selectedResource = undefined;
+    }
+
+    // imposible route source -> source
+    const isSourceOnSuportedDestinations =
+      this.supportedDestinationNetworks.some(
+        (destinationDomain) =>
+          destinationDomain.chainId === this.sourceNetwork?.chainId
+      );
+
+    if (isSourceOnSuportedDestinations) {
```

Comment:
Why not just update filter on L242 instead of copying it here slightly modified?